### DISCUSSION TITLE
IntelliJ IDEA plugin for the slint language server

### DISCUSSION THREAD

**Discussion by megabyte6** [slint-ui/slint#2724] [Ideas] [answered]
Would IntelliJ IDEA ever get a similar plugin to the VSCode extension for slint? I would use it in hand with the rust plugin for IntelliJ. I didn't see a previous discussion about this topic but I apologize if one already exists. Thanks!

**Comment by tronical** [comment]
It would be lovely to have an IntelliJ IDEA plugin for Slint. It would probably be best to re-use as much of the Slint Language server functionality as possible. That kind of re-use capability is something we do keep in the back of our minds as we extend the language server.

**Reply by megabyte6** [reply]
Alright, thanks!

**Comment by hunger** [comment, marked as answer]
Our [README](https://github.com/slint-ui/slint/tree/master/editors#lsp-support) explains how to set up our language server in jetbrains IDEs, using a generic language server plugin. Maybe that can provide you with a clutch for now?

Having a special plugin would of course be much nicer :-)

See below:

@kizeevov did a Intellij plugin [here](https://github.com/kizeevov/slint-idea-plugin).

**Reply by megabyte6** [reply]
Yeah that looks good.
A special plugin would be nice but this works for the moment.
Thanks!

**Comment by bundyo** [comment]
> Our [README](https://github.com/slint-ui/slint/tree/master/editors#lsp-support) explains how to set up our language server in jetbrains IDEs, using a generic language server plugin.

Unfortunately the generic LSP plugin for JetBrains IDE is old, abandoned and therefore incompatible with latest versions.

**Reply by gridlocdev** [reply]
FYI: JetBrains recently added an LSP API for plugin developers as of the IntelliJ IDEA 2023.2 release just a couple of months ago: https://blog.jetbrains.com/platform/2023/07/lsp-for-plugin-developers/

It looks like this is something the Slint LSP could possibly integrate with to create an IntelliJ plugin.

**Reply by ogoffart** [reply]
Thanks for the info. 

If someone do try, let us know and maybe even fill a PR to update the instructions README.

**Reply by kizeevov** [reply]
I really miss the plugin for the IntelliJ IDEA. I started developing [a plugin for Slint](https://github.com/kizeevov/slint-idea-plugin). It's very raw. But the autocomplete is already supported.
Tested on CLion 2023.2 and IDEA Ultimate 2023.2. Not supported RustRover 2023.2 EAP ☹️

**Reply by hunger** [reply]
@kizeevov Oh, cool! may I mention it in our editors README or do you think it is too early for that yet?

**Reply by kizeevov** [reply]
@hunger Yes, of course! I will be glad if you mention it. I plan to continue development. Maybe someone will be interested and help.

**Reply by hunger** [reply]
Is this OK with you? https://github.com/slint-ui/slint/pull/3637

**Reply by kizeevov** [reply]
Yes. Thank you!

**Reply by hunger** [reply]
Pushed! Thanks for the effort! Ping us if you need information or help.
